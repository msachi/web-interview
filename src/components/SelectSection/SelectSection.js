import React from 'react'
import PropTypes from 'prop-types'

import Button from '../Button/Button'

import './SelectSection.scss'

const SelectSection = props => (
  <div className="section">
    <h3 className="section-title">{props.title}</h3>
    <div className="button-wrapper">
      {props.buttons.map(b => (
        <Button
          key={b.title}
          onClick={b.onClick}
          title={b.title}
          active={b.active}
        />
      ))}
    </div>
  </div>
)

SelectSection.propTypes = {
  title: PropTypes.string,
  buttons: PropTypes.arrayOf(PropTypes.Object),
}

export default SelectSection
