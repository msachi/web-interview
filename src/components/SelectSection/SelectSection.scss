@import '../../constants.scss';

.select-section {
  margin-bottom: 24px;

  .header {
    display: flex;
    align-items: center;
    margin-bottom: 20px;
  }

  .icon {
    height: 28px;
    width: 28px;
    margin-right: 10px;
  }

  .title {
    font-weight: bold;
    margin: 0px;
  }

  
  .button-wrapper {
    overflow-x: auto;
    overflow-y: hidden;
    white-space: nowrap;
    margin-left: 30px;
  }
}