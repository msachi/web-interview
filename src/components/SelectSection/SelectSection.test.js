import React from 'react'
import { render, cleanup } from '@testing-library/react'
import SelectSection from './SelectSection'

afterEach(cleanup)

test('Renders SelectSection correctly', () => {
  const container = render(
    <SelectSection
      title="Consultant Type"
      buttons={[
        {
          title: 'GP',
          onClick: () => {},
          active: true,
        },
        {
          title: 'Specialist',
          onClick: () => {},
          active: false,
        },
        {
          title: 'Therapist',
          onClick: () => {},
          active: false,
        },
      ]}
    />
  )
  expect(container).toMatchSnapshot()
})
