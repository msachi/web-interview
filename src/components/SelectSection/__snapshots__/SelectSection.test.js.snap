// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders SelectSection correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="select-section"
      >
        <div
          class="header"
        >
          <img
            alt="Section icon"
            class="icon"
            src="icon.png"
          />
          <h4
            class="title"
          >
            Consultant Type
          </h4>
        </div>
        <div
          class="button-wrapper"
        >
          <button
            class="button button--active"
            type="button"
          >
            GP
          </button>
          <button
            class="button"
            type="button"
          >
            Specialist
          </button>
          <button
            class="button"
            type="button"
          >
            Therapist
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="select-section"
    >
      <div
        class="header"
      >
        <img
          alt="Section icon"
          class="icon"
          src="icon.png"
        />
        <h4
          class="title"
        >
          Consultant Type
        </h4>
      </div>
      <div
        class="button-wrapper"
      >
        <button
          class="button button--active"
          type="button"
        >
          GP
        </button>
        <button
          class="button"
          type="button"
        >
          Specialist
        </button>
        <button
          class="button"
          type="button"
        >
          Therapist
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
